pkg/iptables/service.go

```diff
@@ -0,0 +1,133 @@
+// Package iptables is a wrapper around the iptables binary
+package iptables
+
+import (
+	"errors"
+	"os/exec"
+
+	"github.com/kontainerooo/kontainer.ooo/pkg/abstraction"
+)
+
+// Service handles iptables rules
+type Service interface {
+	// AddRule adds a given iptables rule
+	AddRule(refid uint, rule Rule) error
+	// RemoveRule removes a given iptables rule
+	RemoveRule() error
+	// GetRulesForUser returns a list of all rules for a given user
+	GetRulesForUser(refid uint) []Rule
+	// CreateIPTablesBackup creates an iptables backup file
+	CreateIPTablesBackup() string
+	// LoadIPTablesBackup restores iptables from backup file
+	LoadIPTablesBackup() error
+}
+
+type dbAdapter interface {
+	abstraction.DBAdapter
+	AutoMigrate(...interface{}) error
+	Where(interface{}, ...interface{}) error
+	First(interface{}, ...interface{}) error
+	Find(interface{}, ...interface{}) error
+	Create(interface{}) error
+	Delete(interface{}, ...interface{}) error
+}
+
+type service struct {
+	iptPath string
+	db      dbAdapter
+}
+
+func (s *service) executeIPTableCommand(c string) error {
+	cmd := ExecCommand(s.iptPath, c)
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *service) InitializeDatabases() error {
+	return s.db.AutoMigrate(&iptablesEntry{})
+}
+
+func (s *service) ruleExists(r Rule) bool {
+	s.db.Where("ID = ?", r.GetHash())
+	res := s.db.GetValue()
+	if res != nil {
+		return true
+	}
+	return false
+}
+
+func (s *service) AddRule(refid uint, rule Rule) error {
+	r, err := rule.ToString()
+	if err != nil {
+		return errors.New("Not a valid rule")
```

Comment:
Wouldn't it be better to return the error returned by the ToString function since it tells the exact problem with the rule?